test/tuple.cpp

```diff
@@ -13,9 +13,47 @@
 //
 //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
 
+#include <type_traits>
+
 #include "camp/camp.hpp"
 #include "gtest/gtest.h"
 
+static_assert(std::is_same<camp::tuple<int &, int const &, int>,
+                           decltype(camp::tuple_cat_pair(
+                               camp::val<camp::tuple<int &>>(),
+                               camp::val<camp::tuple<int const &, int>>()))>::value,
+              "tuple_cat pair nuking refs");
+
+// Size tests, ensure that EBO is being applied
+struct A {
+};
+struct B {
+};
+
+static_assert(sizeof(camp::tuple<A, B>) == 1, "EBO working as intended with empty types");
+
+static_assert(sizeof(camp::tuple<A, B, ptrdiff_t>) == sizeof(ptrdiff_t),
+              "EBO working as intended with one sized type at the end");
+
+static_assert(sizeof(camp::tuple<A, B, ptrdiff_t>) == sizeof(ptrdiff_t),
+              "EBO working as intended with one sized type in the middle");
```

Comment:
```suggestion
// These are off by default, because failing this is not a fatal condition
#ifdef TEST_EBO
static_assert(sizeof(camp::tuple<A, B>) == 1, "EBO working as intended with empty types");

static_assert(sizeof(camp::tuple<A, B, ptrdiff_t>) == sizeof(ptrdiff_t),
              "EBO working as intended with one sized type at the end");

static_assert(sizeof(camp::tuple<A, ptrdiff_t, B>) == sizeof(ptrdiff_t),
              "EBO working as intended with one sized type in the middle");
#endif // TEST_EBO
```